src/scripts/main.js

```diff
@@ -1,3 +1,81 @@
 'use strict';
 
-// write code here
+const baseUrl = `https://mate-academy.github.io/`
++ `phone-catalogue-static/api/phones`;
+
+function request(url) {
+  return fetch(url)
+    .then(responce => responce.json());
+}
+
+const phonesIdListArray
+  = request(`${baseUrl}.json`)
+    .then(result => result)
+    .then(arrayPhonesResult => {
+      return arrayPhonesResult.map(phone => {
+        return request(`${baseUrl}/${phone.id}.json`);
+      });
+    });
+
+function getFirstReceivedDetails(arrayPhonesId) {
+  return arrayPhonesId
+    .then(AllDetailsResult => Promise.any(AllDetailsResult));
```

Comment:
```suggestion
    .then(AllDetailsResult => Promise.any(AllDetailsResult));
```
Нам потрібен перший результат, а не будь який. Для цього підійде Promise.race